### ISSUE TITLE
Suggestion: add support for b_lto=optional

### ISSUE THREAD

**Issue by hdante** [mesonbuild/meson#7387]
Hello, for supporting projects providing third-party libraries, meson should support optional LTO. This is possible, for example, using gcc -flto -ffat-lto-objects with static libraries, where gcc generates both GIMPLE and native object code output. Optional LTO is necessary for distributing a library because LTO output formats are incompatible among toolchains and distributing a LTO linked library would lock the user to a specific compiler family.

Behavior: if supported by the toolchain, the build system generates both (toolchain specific) LTO object and a platform compatible object. If not supported by the toolchain, the build system only generates the platform compatible object. Result: users can use any compiler, by default not having LTO support. When using the appropriate toolchain, user will have LTO.

**Comment by thesamesam** [comment]
I agree we should do something here. Let's call it a dupe of your other bug (https://github.com/mesonbuild/meson/issues/10300) where there's some more discussion.
